### ISSUE TITLE
 Hide other variants’ media after selecting a variant

### ISSUE THREAD

**Issue by Miktau** [Shopify/dawn#456]
**Describe the current behavior**

**Describe the expected behavior**

**Version information (Dawn, browsers and operating systems)**
- Dawn Version: 1.0.0
- Chrome Version 91.0.4472.114
- macOS Version 11.3.1

**Possible solution**

**Additional context/screenshots**
Add any other context about the problem here. If applicable, add screenshots to help explain.

**Comment by gregjotau** [comment]
Hi @ludoboludo , I see you commited this change

We have the code on famme.no, but it seems to not work. I see the logic:

section.settings.hide_variants and variant_images contains media.src

I guess there is some other condition we are misunderstanding? How is this supposed to work and hide variant images?

![image](https://user-images.githubusercontent.com/13047497/131576050-8d2d6c75-b9f7-44cd-9df2-f29569c4ea0d.png)

**Comment by ludoboludo** [comment]
Hey @gregjotau, 

Since you can only associate one image to a variant at the moment, the way it works is that it hides any media that is associated with a variant except the first one. And on variant change, it will add the image/media associated with that new variant at the beginning of the list of images. 

`updateMedia()` will take care of adding that variant media at the top of the list while we only show the first element with the class `product__media-item--variant`.

Let us know where it's not working/meeting your expectation so we can see if something isn't working as intended. 

Thanks for bringing this up.

**Comment by gregjotau** [comment]
OK, so one thing I would expect:

1. Variants are added in shopify backend chronologically
2. Every image from the selected, until the next image connected to a variant

Since only one image can be associated to a variant, it is very hard to implement anything that makes sense. I do not understand really how we could actually show just black leggings if one chooses the black variant with the current system?

**Comment by ludoboludo** [comment]
> OK, so one thing I would expect:
> 
> 1. Variants are added in shopify backend chronologically
> 2. Every image from the selected, until the next image connected to a variant
> 
> Since only one image can be associated to a variant, it is very hard to implement anything that makes sense. I do not understand really how we could actually show just black leggings if one chooses the black variant with the current system?

Yeah this is something missing that would be a nice feature from the admin. The possibility to link multiple medias to one variant.

`Since only one image can be associated to a variant, it is very hard to implement anything that makes sense.` very true. Hopefully this is something that will change down the line.

**Comment by ludoboludo** [comment]
I don't know about soon nor timeframe in general. This will be up to the product team to determine the best course of action and timeframe to work on this and get it done.

**Comment by Pixelous** [comment]
@ludoboludo hi, I was surprised as well that you can not assign multiple images for a variant, can you please ask product team when they will add this feature?

**Comment by ludoboludo** [comment]
Hey @Pixelous 👋 , I don't have any ETA to provide at the moment but they're aware of the need for it 👍

**Comment by tydonk** [comment]
Why was this issue closed?  It does not work.  I have 42 variants of my product with 7 photos each, that makes 294 images that load.  It's impossible for a customer to navigate this and looks terrible.  Is there an updated timeline on fixing this?  The last comment was almost 2 years ago.
